### PR TITLE
fix(emp-tools): change cors proxy and cleanup other requests

### LINKED ISSUE #205: cors errors reaching bitstamp price api
image
It looks like the cors proxy we were using expired: `https://cors-anywhere.herokuapp.com/https://www.bitstamp.net/api/v2/ticker/btcusd`

@mrice32 was able to spin up our own on gcp. need to update emp-tools to use this. one small caveat is it wont accept the protocol part of the url, so https will need to be stripped before proxying.

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

motivation #205 

Simply adds new cors proxy wrapper to bitstamp url. Adds as helper function in case we need to do this for other urls as well. 
Also updates coingecko calls to remove uncessessary header info, did not run into issues after this change, but verify if you can. 

fixes #205

### CODE REVIEW

File: utils/getCoinGeckoTokenPrice.ts [hunk 1]
```diff
@@ -3,17 +3,15 @@
...
 import fetch from "isomorphic-fetch";
+const fetchOptions = {
+  // currently no options needed, previously we were setting content type and accept.
+};
 
 export async function getUmaPrice() {
   const query =
     "https://api.coingecko.com/api/v3/simple/price?ids=uma&vs_currencies=usd";
 
-  const response = await fetch(query, {
```
Comment (mrice32) [inline utils/getCoinGeckoTokenPrice.ts:11]: Why remove these? Did they not work with these options? Or is this just not required anymore?

File: utils/getOffchainPrice.ts [hunk 2]
```diff
@@ -30,6 +30,19 @@ function _getBitstampPriceFromJSON(jsonData: any) {
...
 
+// This is needed because our new cors proxy wont forward http or https. We really only use https.
+function stripProtocol(url: string) {
+  return url.replace("https://", "");
+}
+
+// Small helper to wrap a url in cors proxy.
+function proxyUrl(
+  url: string,
+  proxy = "https://us-central1-uma-protocol.cloudfunctions.net/cors-proxy"
```
Comment (mrice32) [inline utils/getOffchainPrice.ts:41]: Long-term, we should probably have this proxy address passed in as a build env variable.

**Review by mrice32 — APPROVED** [review]
LGTM!

File: utils/getCoinGeckoTokenPrice.ts (hunk 1 above)
Comment (daywiss) [reply utils/getCoinGeckoTokenPrice.ts:11]: While testing out this cors stuff, i removed the headers and did not notice any detrimental effects, It was working fine with them, but it just seemed unnecessary so I left them out.

File: utils/getOffchainPrice.ts (hunk 2 above)
Comment (daywiss) [reply utils/getOffchainPrice.ts:41]: not sure if this project will reference a .env, didnt try. If it does though it should be easy to add

**Review by nicholaspai — APPROVED** [review]
LGTM

File: utils/getCoinGeckoTokenPrice.ts [hunk 3]
```diff
@@ -3,17 +3,18 @@
 
 // so this can be tested outside browser
 import fetch from "isomorphic-fetch";
+const fetchOptions = {
+  headers: {
```
Comment (daywiss) [inline utils/getCoinGeckoTokenPrice.ts:7]: i added the header options back in. rather not introduce potential bug.

File: utils/getOffchainPrice.ts [hunk 4]
```diff
@@ -30,6 +30,22 @@ function _getBitstampPriceFromJSON(jsonData: any) {
...
+function stripProtocol(url: string) {
+  return url.replace("https://", "");
+}
+
+// Small helper to wrap a url in cors proxy.
+export function proxyUrl(
+  url: string,
+  // This env can be set with a .env file. It has to be prefixed with NEXT_PUBLIC_ to be available in browser
+  // This has a default set for now until we can get prod env var in.
+  proxy = process.env.NEXT_PUBLIC_CORS_PROXY_URL ||
```
Comment (daywiss) [inline utils/getOffchainPrice.ts:43]: i tested this. and it does inject env in if set. left in hardcoded string so it works in production

File: utils/getOffchainPrice.ts (hunk 2 above)
Comment (daywiss) [reply utils/getOffchainPrice.ts]: added in optional env var. see comment

File: utils/getCoinGeckoTokenPrice.ts (hunk 1 above)
Comment (daywiss) [reply utils/getCoinGeckoTokenPrice.ts:11]: added this back in just in case

**Pull request merged**